src/realtime_pollen_calibration/utils.py

```diff
@@ -1 +1,552 @@
+        tune_stns = get_field_at(
+            ds,
+            pollen_type + "tune",
+            coord_stns[istation],
+        )
+        # saison days at the current station
+        # if > 0 then the pollen season has started
+        saisn_stns = get_field_at(ds, pollen_type + "saisn", coord_stns[istation])
+        if verbose:
+            print(
+                f"Current station n°{istation}, ",
+                f"(lat: {coord_stns[istation][0]}, ",
+                f"lon: {coord_stns[istation][1]}), ",
+                f"last 120H concentration observed: {sum_obs}, ",
+                f"modeled: {sum_mod}",
+            )
+            print(
+                f"Current tune value {tune_stns.values[0][0]} ",
+                f"and saisn: {saisn_stns.values[0][0]}",
+            )
+            print("-----------------------------------------")
+        if (saisn_stns > 0) and ((sum_obs <= 720) or (sum_mod <= 720)):
+            if verbose:
+                print(
+                    "Season started but low observation or modeled concentrations, "
+                    "(tune)**(-1/24) = "
+                    f"{(tune_pol_default / tune_stns.values[0][0]) ** (1 / 24)}"
+                )
+            change_tune[istation] = (tune_pol_default / tune_stns) ** (1 / 24)
+        if (saisn_stns > 0) and (sum_obs > 720) and (sum_mod > 720):
+            if verbose:
+                print(
+                    "Season started and high observation ", "and modeled concentrations"
+                )
+            change_tune[istation] = (sum_obs / sum_mod) ** (1 / 24)
+        if verbose:
+            print(f"Change tune is now: {change_tune[istation]}")
+            print("-----------------------------------------")
+    return change_tune
+
+
+def get_change_phenol(
+    pollen_type, array_obs, ds, coord_stns, verbose=False
+):  # pylint: disable=R0912,R0914,R0915
+    """Compute the change of the temperature thresholds for the phenology.
+
+    Args:
+        pollen_type: String describing the pollen type analysed.
+        array_obs: Last 120H pollen concentration observed.
+        ds: xarray.DataSet containing 'T_2M', 'tthrs', 'tthre'
+            (for POAC, 'saisl' instead), 'saisn' and 'ctsum'.
+        coord_stns: List of (lat, lon) tuples of the stations
+        verbose: Optional additional debug prints.
+
+    Returns:
+        change_tthrs: Amount by which tthrs should be changed at each station
+        change_tthre_saisl: As above for tthre or saisl for POAC.
+    The new threshold value corresponds to:
+    tthrs(station, T+dT) = tthrs(station, T) + change_tthrs(station).
+
+    """
+    date = pd.Timestamp(ds.time.values).day_of_year + 1 + 31
+    thr_con_24 = {"ALNU": 240, "BETU": 240, "POAC": 72, "CORY": 240}
+    thr_con_120 = {"ALNU": 720, "BETU": 720, "POAC": 216, "CORY": 720}
+    failsafe = {"ALNU": 1000, "BETU": 2500, "POAC": 6000, "CORY": 2500}
+    jul_days_excl = {"ALNU": 14, "BETU": 40, "POAC": 3, "CORY": 46}
```

Comment:
agree. However, values that occur just once are a minor problem. A problem is if the same value has to be changed in 10 different places. This would be error-prone

Reply:
I just moved them at the top of the utils.py file, out of the subroutines. Maybe you will want to move them to a kind of config file and simply load them in the utils.py file but at least now they are only defined once and at the top of a file so they are easy to modify.